app/models/template_input.rb

```diff
@@ -91,11 +91,12 @@ def resolved_value
 
   class UserInputResolver < InputResolver
     def ready?
-      !input_value.nil?
+      @renderer.invocation
     end
 
     def resolved_value
-      input_value.value
+      raise(ValueNotReady, _("Required input not specified")) if @input.required? && input_value.try(:value).blank?
```

Comment:
So imagine "yum custom options" that could contains things like -y or --nogpgcheck but does not have to contain anything hence empty string. If I understand your change correctly, if I want to rely on input to exist, the input value must exist and valid value is also empty string. Otherwise I'd not be able to write template like `yum install sshd <%= input('custom_args') %>`, if it's not required, the input would never be ready